routing/datasource.go

```diff
@@ -389,12 +394,12 @@ func receiveRouteMatcher(o Options, out chan<- *matcher, quit <-chan struct{}) {
 			for _, err := range errs {
 				o.Log.Error(err)
 			}
-
-			mout = m
+			// TODO: What about the errors? Should we exclude the routes which has errors?
```

Comment:
as i understand, there is the syntax validation and then the semantic to a certain extent, e.g. the reference to a missing filter. How the users would see it from the outside, i believe: i put in these routes, and i want to see which ones were applied, which ones failed. So yes, i think we need to consider both kinds of failures.